iati/tests/test_resources.py

```diff
@@ -50,193 +177,492 @@ class TestResourceFolders(object):
+        minor_version = max(iati.version.versions_for_integer(std_ver_major_uninst_valid_known))
 
-        content = iati.utilities.load_as_string(path)
+        major_path = func_to_test(filename_no_meaning_single, std_ver_major_uninst_valid_known)
+        minor_path = func_to_test(filename_no_meaning_single, minor_version)
 
-        assert len(content) > 5000
-        assert iati.validator.is_xml(content)
+        assert major_path == minor_path
 
+    def test_create_path_no_version(self, filename_no_meaning_single, func_to_test):
+        """Check that specifying a version of the Standard to create a path for is required."""
+        with pytest.raises(TypeError):
+            func_to_test(filename_no_meaning_single)
 
-class TestResourceRulesets(object):
-    """A container for tests relating to Ruleset resources."""
+    def test_create_path_unknown(self, filename_no_meaning_single, std_ver_all_mixedinst_valid_unknown, func_to_test):
+        """Check that a ValueError is raised when using a generation function to create a path for a at an unknown version of the Standard."""
+        with pytest.raises(ValueError):
+            func_to_test(filename_no_meaning_single, std_ver_all_mixedinst_valid_unknown)
 
-    def test_get_ruleset_paths(self, std_ver_minor_mixedinst_valid_fullsupport):
-        """Check that all ruleset paths are found."""
-        ruleset_paths = iati.resources.get_ruleset_paths(std_ver_minor_mixedinst_valid_fullsupport)
+    def test_create_path_ver_typerr(self, filename_no_meaning_single, std_ver_all_uninst_typeerr, func_to_test):
+        """Check that a TypeError is raised when using a generation function to create a path from a version of an incorrect type."""
+        with pytest.raises(TypeError):
+            func_to_test(filename_no_meaning_single, std_ver_all_uninst_typeerr)
 
-        assert len(ruleset_paths) == 1
+    def test_create_path_path_valueerr(self, filepath_invalid_value, std_ver_minor_inst_valid_single, func_to_test):
+        """Check that a ValueError is raised when providing a generation function a path to work from that is a string that cannot be a filepath."""
+        with pytest.raises(ValueError):
+            func_to_test(filepath_invalid_value, std_ver_minor_inst_valid_single)
 
+    def test_create_path_path_typeerr(self, filepath_invalid_type, std_ver_minor_inst_valid_single, func_to_test):
+        """Check that a TypeError is raised when providing a generation function a path to work from that is of a type that cannot be a filepath."""
+        with pytest.raises(TypeError):
+            func_to_test(filepath_invalid_type, std_ver_minor_inst_valid_single)
 
-class TestResourceSchemas(object):
-    """A container for tests relating to Schema resources."""
 
-    def test_get_activity_schema_paths(self, std_ver_minor_mixedinst_valid_fullsupport):
-        """Check that all activity schema paths are found.
+@pytest.mark.new_tests
+class TestResourceGetCodelistMappingPaths(object):
```

Comment:
Some of these class names become rather clunky. Info like `Resource` in the name is fairly redundant (it's in the `test_resources` module!), and could be provided in a more scalable manner using `@pytest.mark.resources` instead.